goyacc/format_yacc.go

```diff
@@ -39,9 +39,7 @@ func Format(inputFilename string, goldenFilename string) (err error) {
 	if err = yFmt.Setup(goldenFilename); err != nil {
 		return err
 	}
-	defer func() {
-		err = yFmt.Teardown()
-	}()
```

Comment:
why change this? here will store the err about Teardown() failure

Reply:
If we store the error of `Teardown()`, the original error will be overridden.

Reply:
maybe only replace error when it was nil.

```go
teardownErr := yFmt.Teardown()
if err == nil {
    err = teardownErr
}
```